javalanglib/src/main/scala/java/lang/Integer.scala

```diff
@@ -81,20 +89,37 @@ object Integer {
         }
         val res = js.Dynamic.global.parseInt(s, radix).asInstanceOf[scala.Double]
 
-        if (res.isNaN || res > MAX_VALUE || res < MIN_VALUE)
+        if (res.isNaN ||
+          (signed && (res > MAX_VALUE || res < MIN_VALUE)) ||
+          (!signed && (res > 0xFFFFFFFFL || res < 0))) {
```

Comment:
Besides, the condition on 3 lines is too long anyway. I recommend putting the bounds checks in a local `@inline def`. 
Also consider writing the bounds checks based on an `if (signed) ... else ...`. The intent would be clearer.